File: src/main/java/net/starkenberg/logging/elastic/ElasticMaintenanceApplication.java
package net.starkenberg.logging.elastic;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.web.client.RestTemplate;

/**
 * @author Brad Starkenberg
 * main application class
 */
@SpringBootApplication
@EnableScheduling
@EnableConfigurationProperties(ConfigProps.class)
public class ElasticMaintenanceApplication {

    public static void main(String[] args) {
        SpringApplication.run(ElasticMaintenanceApplication.class, args);
    }
    
    @Bean(name="restTemplate")
    public RestTemplate restTemplate() {
    	return new RestTemplate();
    }
    
}

File: src/main/java/net/starkenberg/logging/elastic/ElasticMaintenanceApplication.kt
package net.starkenberg.logging.elastic

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.ComponentScan
import org.springframework.scheduling.annotation.EnableScheduling

/**
 * @author Brad Starkenberg
 * main application class
 */
@EnableAutoConfiguration
@ComponentScan
@EnableScheduling
@EnableConfigurationProperties(ConfigProps::class)
class ElasticMaintenanceApplication {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            SpringApplication.run(ElasticMaintenanceApplication::class.java, *args)
        }
    }

}
